ogre2/src/Ogre2ParticleEmitter.cc

```diff
@@ -413,7 +428,50 @@ void Ogre2ParticleEmitter::SetColorRangeImage(const std::string &_image)
 void Ogre2ParticleEmitter::Init()
 {
   Ogre2Visual::Init();
+  this->CreateParticleSystem();
+}
+
+//////////////////////////////////////////////////
+void Ogre2ParticleEmitter::PreRenderImpl()
+{
+  // \todo(anyone) rename this function to PreRender() so it overrides function
+  // from base class
+
+  // recreate the particle system if needed
+  // currently this is needed when user changes type or particle size
+  if (this->dataPtr->emitterDirty)
+  {
+    this->Destroy();
+    this->CreateParticleSystem();
 
+    // make direct ogre calls here so we don't mark emitter as dirty again
+    this->dataPtr->ps->setDefaultDimensions(
+        this->particleSize[0], this->particleSize[1]);
+
+    this->SetEmitterSize(this->emitterSize);
+
+    // set other properties
```

Comment:
I believe we need to call `SetDuration` and `SetEmitting` here as well, since these methods set properties for `this->dataPtr->emitter`.

Reply:
https://github.com/ignitionrobotics/ign-rendering/pull/241/commits/a19125e60dafbc01f48f882465cda6b3b5615919